core/src/main/java/org/apache/iceberg/MetricsSpec.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Maps;
+import java.util.Locale;
+import java.util.Map;
+
+import static org.apache.iceberg.TableProperties.DEFAULT_WRITE_METRICS_MODE;
+import static org.apache.iceberg.TableProperties.DEFAULT_WRITE_METRICS_MODE_DEFAULT;
+
+public class MetricsSpec {
+  public enum MetricsMode {
+    FULL,
+    COUNTS,
+    NONE;
+
+    static MetricsMode fromString(String mode) {
+      Preconditions.checkArgument(mode != null, "Cannot convert null into a metrics mode");
+      return MetricsMode.valueOf(mode.toUpperCase(Locale.ENGLISH));
+    }
+  }
+
+  private static final String COLUMN_CONF_PREFIX = "write.metadata.metrics.column.";
+
+  private Map<String, MetricsMode> columnModes = Maps.newHashMap();
+  private MetricsMode defaultMode;
+
+  private MetricsSpec() {}
+
+  public static MetricsSpec getDefault() {
+    MetricsSpec spec = new MetricsSpec();
+    spec.defaultMode = MetricsMode.FULL;
+    return spec;
+  }
+
+  public static MetricsSpec fromProperties(Map<String, String> props) {
+    MetricsSpec spec = new MetricsSpec();
+    props.keySet().stream()
+        .filter(key -> key.startsWith(COLUMN_CONF_PREFIX))
+        .forEach(key -> {
+          MetricsMode mode = MetricsMode.fromString(props.get(key));
+          String columnAlias = key.replaceFirst(COLUMN_CONF_PREFIX, "");
+          spec.columnModes.put(columnAlias, mode);
+        });
+    String defaultModeAsString = props.getOrDefault(DEFAULT_WRITE_METRICS_MODE, DEFAULT_WRITE_METRICS_MODE_DEFAULT);
+    spec.defaultMode = MetricsMode.fromString(defaultModeAsString);
+    return spec;
+  }
+
+  public MetricsMode getColumnMetricsMode(String columnAlias) {
```

Comment:
This is a long name and we typically leave out `get` because it doesn't provide much value. What about `metricsSpec.columnMode(name)`?